test/algorithms/set_operations/intersection/intersection_areal_areal_linear.cpp

```diff
@@ -0,0 +1,318 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+// Unit test
+
+// Copyright (c) 2015, Oracle and/or its affiliates.
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+// Contributed and/or modified by Menelaos Karavelas, on behalf of Oracle
+
+#ifndef BOOST_TEST_MODULE
+#define BOOST_TEST_MODULE test_intersection_linear_linear_areal
+#endif
+
+#ifdef BOOST_GEOMETRY_TEST_DEBUG
+#define BOOST_GEOMETRY_DEBUG_TURNS
+#define BOOST_GEOMETRY_DEBUG_SEGMENT_IDENTIFIER
+#endif
+
+#include <boost/test/included/unit_test.hpp>
+
+#include <boost/range.hpp>
+
+#include <boost/geometry/core/closure.hpp>
+#include <boost/geometry/core/point_type.hpp>
+
+#include <boost/geometry/geometries/linestring.hpp>
+#include <boost/geometry/geometries/multi_linestring.hpp>
+#include <boost/geometry/geometries/ring.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/geometry/geometries/multi_polygon.hpp>
+
+#include "test_intersection_linear_linear.hpp"
+
+typedef bg::model::point<double,2,bg::cs::cartesian>  point_type;
+typedef bg::model::multi_linestring
+    <
+        bg::model::linestring<point_type>
+    >  multi_linestring_type;
+
+typedef bg::model::ring<point_type, true, false> open_ring_type;
+typedef bg::model::polygon<point_type, true, false> open_polygon_type;
+typedef bg::model::multi_polygon<open_polygon_type> open_multipolygon_type;
+
+
+template <typename Areal, typename Tag = typename bg::tag<Areal>::type>
+struct closed_type
+{};
+
+template <typename Ring>
+struct closed_type<Ring, bg::ring_tag>
+{
+    typedef bg::model::ring
+        <
+            typename bg::point_type<Ring>::type, true, false
+        > type;
+};
+
+template <typename Polygon>
+struct closed_type<Polygon, bg::polygon_tag>
+{
+    typedef bg::model::polygon
+        <
+            typename bg::point_type<Polygon>::type, true, false
+        > type;
+};
+
+template <typename MultiPolygon>
+struct closed_type<MultiPolygon, bg::multi_polygon_tag>
+{
+    typedef bg::model::multi_polygon
+        <
+            typename closed_type
+                <
+                    typename boost::range_value<MultiPolygon>::type
+                >::type
+        > type;
+};
```

Comment:
I doubt if we should do such amount of meta-programming just in unit-tests.
Are there not more straightforward solutions?
Why not specify Open/Close or CW/CCW as a template argument in the main test function, and call correct (as I usually do)? That is quite simple. This asks for putting it in a headerfile, share it, and you get very complex tests which in the end do the same thing.